lib/handlers/ntlm.ts

```diff
@@ -146,12 +128,30 @@ export class NtlmCredentialHandler implements ifm.IRequestHandler {
             throw new Error('www-authenticate not found on response of second request');
         }
 
-        const type2msg = ntlm.parseType2Message(res.message.headers['www-authenticate']);
-        const type3msg = ntlm.createType3Message(type2msg, this._ntlmOptions);
+        /**
+         * Server will respond with challenge/nonce
+         * assigned to response's "WWW-AUTHENTICATE" header
+         * and should adhere to RegExp /^NTLM\s+(.+?)(,|\s+|$)/
+         */
+        const serverNonceRegex: RegExp = /^NTLM\s+(.+?)(,|\s+|$)/;
+        const serverNonce: Buffer = Buffer.from(
```

Comment:
What happens here if the server sends back a bad response? It looks like the decodeType2 function fails, we might want to catch that and throw a slightly nicer error (something like `decoding server's type 2 message failed with error: [original message here]`)

Reply:
Yeah, one should intercept bad scenarios as well 👍 .. Will throw an error here to reflect such issue